core/src/main/java/org/apache/iceberg/util/ArrayUtil.java

```diff
@@ -320,4 +320,29 @@ public static boolean isStrictlyAscending(long[] array) {
 
     return true;
   }
+
+  @SuppressWarnings("unchecked")
+  public static <T> T[] concat(Class<T> type, T[]... arrays) {
```

Comment:
for reference, whats the reason for not using library like apache commons?

Reply:
We are trying to avoid an explicit dependency on commons. Guava has a method to merge two but it does not support an array of arrays and null values as valid elements.

Reply:
I should probably add Javadoc on handling null arrays here.